### PR TITLE
To Do list: interactive list

### PR DESCRIPTION
**Pull request by Obimbo07**
This pull request introduces the following features:

1. Methods related to updates 
2. Add event listener to the checkbox (change).
3. Update items object's value for completed key upon user actions.
4. Implement a function for the "Clear all completed" button (use filter() method).
5. Store the updated array of items in local storage,

### CODE REVIEW

File: src/modules/do.js
Comment (Efiamotu-1) [inline src/modules/do.js:1]: - [ ] Your readme has broken links which are unprofessional for a professional readme file, Your links to your twitter account and the live demo of your project are broken please kindly fix them and provide the appropriate links 👍

Comment (Efiamotu-1) [inline src/modules/do.js:1]: - [x] There is a bit of defect with regards to the functionality of your todo which I would like you to fix, upon clearing all the todo using the clear all completed button, it gets deleted on screen and even on the local storage but when i try adding a new task all of the previously deleted tasks reappear both on the screen and on the localstorage. Please kindly work on fixing this issue
- [ ] You have also not implemented the functionality of deleting a single task, there is no delete button when a task is selected. this is a requirement from the previous milestone so this is optional but i would strongly advise that you add  it to your project. Thanks 👍

**Review by Efiamotu-1 — CHANGES_REQUESTED** [review]
Hi @Obimbo07 , 

 

Good job so far! 👏. Your Code is Awesome! 💪

There are some **few** issues 🐛 that you still need to work 🔧 on though to go to the next project ⏭️ but you are almost there! 🏃
## Don't Worry 🙏

It's ok to have bugs in your code 🐛. Please, see this code review as a **chance** for you to **learn** practical coding skills and learn **new tips** that can improve your code quality 💯 and help you **get a better ** 💼.

_I am going to guide you on how to fix them down below. 🤓_

## Highlights :dart: 
- ✔️ No linter Errors
- ✔️ Descriptive README
- ✔️ Descriptive PR title and description
- The code is top-notch! 💯 and has a huge potential.

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._


## Important Notes 📝

- Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
- Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

> **Note:** Please, don't Merge This Pull Request yet. Only Merge when it is Approved ✅.


## Thank you for your time 🙏 and call me if you need help or clarification ☎️

- This is My **GitHub**: [@Efiamotu-1](https://github.com/Efiamotu-1/)
- This is My **Twitter**: [@EFYAMOTU](https://twitter.com/EFYAMOTU)
- This is My **linkedIn**.[Habeeb Efiamotu Musa](https://www.linkedin.com/in/musa-habeeb/)
- This is My **Slack**: [Habeeb Efiamotu Musa](https://slack.com)

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by Obimbo07** [comment]
Please consider my review as I fix the issue about the local storage as instructed by the previous reviewer

File: src/modules/do.js
Comment (Lizdev-05) [inline src/modules/do.js:1]: - [ ] It seems that the issue mentioned by the previous reviewer has not been resolved, as evidenced in the video below. Despite using the "clear all completed" button to remove all tasks, they still reappear both on the screen and in local storage when a new task is added. I kindly request that you fix this issue. :pray: 
  
  [Screencast from 08-19-2023 02:04:51 PM.webm](https://github.com/Obimbo07/To-do-list/assets/73854999/4443c5d6-31dc-4fa7-9068-95e4e779c40c)

**Review by Lizdev-05 — CHANGES_REQUESTED** [review]
Hi @Obimbo07 , 

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.



 ## Highlights
:heavy_check_mark: Professional Readme file :+1: 
:heavy_check_mark: Good commit history :+1:
:heavy_check_mark: No linter error :+1: 

 ## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by Obimbo07** [comment]
Hello reviewer please consider my approval for the project, please help include the local storage error on the display of the clear completed tasks I will implement it but please help me achieve my deadline for project submission

**Review by Qoosim — APPROVED** [review]
#  *Status: Approved* :tada: :heavy_check_mark: 

Hi @Obimbo07, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights
-  Implemented required changes. :heavy_check_mark: 
-  App works as expected. :heavy_check_mark: 
-  No linter errors. :heavy_check_mark: 
-  Good README file. :heavy_check_mark: 
-  Correct GitHub Flow. :heavy_check_mark: 



## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**